app/src/main/java/com/andrew/tasky/agenda/presentation/adapters/PhotoItemAdapter.kt

```diff
@@ -1,81 +1,101 @@
 package com.andrew.tasky.agenda.presentation.adapters
 
 import android.content.Context
-import android.net.Uri
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import com.andrew.tasky.agenda.domain.models.EventPhoto
+import com.andrew.tasky.agenda.util.UiEventPhoto
 import com.andrew.tasky.databinding.ItemPhotoAdapterCardBinding
 import com.bumptech.glide.Glide
 
 class PhotoItemAdapter(
     val context: Context,
     private val onPhotoClick: (Int) -> Unit,
     private val onAddPhotoClick: () -> Unit,
-    private val userIsAttendee: Boolean
-) : ListAdapter<EventPhoto, PhotoItemAdapter.PhotoItemViewHolder>(Companion) {
+) : ListAdapter<UiEventPhoto, RecyclerView.ViewHolder>(Companion) {
 
-    companion object : DiffUtil.ItemCallback<EventPhoto>() {
-        override fun areItemsTheSame(oldItem: EventPhoto, newItem: EventPhoto): Boolean {
-            return oldItem.key == newItem.key
-        }
+    companion object : DiffUtil.ItemCallback<UiEventPhoto>() {
+        private const val TYPE_ADD_PHOTO = 0
+        private const val TYPE_EVENT_PHOTO = 1
 
-        override fun areContentsTheSame(oldItem: EventPhoto, newItem: EventPhoto): Boolean {
-            return if (oldItem is EventPhoto.Local && newItem is EventPhoto.Local) {
-                oldItem.uri == newItem.uri
-            } else if (oldItem is EventPhoto.Remote && newItem is EventPhoto.Remote) {
-                oldItem.photoUrl == newItem.photoUrl
+        override fun areItemsTheSame(oldItem: UiEventPhoto, newItem: UiEventPhoto): Boolean {
+            return if (oldItem is UiEventPhoto.Photo && newItem is UiEventPhoto.Photo) {
+                oldItem.eventPhoto.key == newItem.eventPhoto.key
+            } else if (oldItem is UiEventPhoto.AddPhoto && newItem is UiEventPhoto.AddPhoto) {
+                oldItem == newItem
             } else false
         }
-    }
 
+        override fun areContentsTheSame(oldItem: UiEventPhoto, newItem: UiEventPhoto): Boolean {
+            return if (oldItem is UiEventPhoto.Photo && newItem is UiEventPhoto.Photo) {
+                return if (oldItem.eventPhoto is EventPhoto.Local &&
+                    newItem.eventPhoto is EventPhoto.Local
+                ) {
+                    oldItem.eventPhoto.uri == newItem.eventPhoto.uri
+                } else if (oldItem.eventPhoto is EventPhoto.Remote &&
+                    newItem.eventPhoto is EventPhoto.Remote
+                ) {
+                    oldItem.eventPhoto.photoUrl == newItem.eventPhoto.photoUrl
+                } else false
+            } else oldItem is UiEventPhoto.AddPhoto && newItem is UiEventPhoto.AddPhoto
+        }
+    }
```

Comment:
Is every subclass of `UiEventPhoto` a data class? If so, you could just `return oldItem == newItem`